src/server/dflycmd.cc

```diff
@@ -475,11 +508,13 @@ OpStatus DflyCmd::StartFullSyncInThread(FlowInfo* flow, Context* cntx, EngineSha
     flow->saver.reset();
   };
 
-  sf_->journal()->StartInThread();
-
   // Shard can be null for io thread.
   if (shard != nullptr) {
-    flow->saver->StartSnapshotInShard(true, cntx->GetCancellation(), shard);
+    if (flow->start_partial_sync_at < UINT64_MAX)
```

Comment:
Since you're repeating this condition, maybe introduce some `IsPartialSync()` or `kInvalidPartialSync`?

Reply:
changed to using optional